packages/amplify-gen2-codegen/src/backend/synthesizer.ts

```diff
@@ -557,13 +558,19 @@ export class BackendSynthesizer {
       factory.createVariableDeclarationList([backendVariable], ts.NodeFlags.Const),
     );
 
+    let envVariableIdentifier = 'process.env.AMPLIFY_GEN_1_ENV_NAME';
+    if (ci.isCI && !process.env.AMPLIFY_GEN_1_ENV_NAME) {
```

Comment:
should the CI check be part of the codegen so that it can be evaluated at runtime in `backend.ts` file ?